esrally/reporter.py

```diff
@@ -34,6 +34,26 @@ def print_internal(message):
 def print_header(message):
     print_internal(console.format.bold(message))
 
+def write_single_report(self, report_file, headers, data, write_header=True, show_also_in_console=True):
```

Comment:
`self` is an (implicit) reference to the current object. As you refactored it to be a function you need to remove `self` here and also move all related helper functions (like `format_as_markdown`, `format_as_csv`).